pandas/indexes/multi.py

```diff
@@ -1638,6 +1640,18 @@ def maybe_droplevels(indexer, levels, drop_level):
         if isinstance(key, list):
             key = tuple(key)
 
+        # must be lexsorted to at least as many levels as the level parameter,
+        # or the number of items in the key tuple.
+        # Note: level is 0-based
+        required_lexsort_depth = level + 1
+        if isinstance(key, tuple):
+            required_lexsort_depth = max(required_lexsort_depth, len(key))
+        if self.lexsort_depth < required_lexsort_depth:
+            raise KeyError('MultiIndex Slicing requires the index to be '
```

Comment:
I think this error message could be clearer, something like:

> 'MultiIndex slicing requires the index to be fully lexsorted up to the {required}th level, lexsort depth is currently {current}'

Just make it clearer which number is the required one and which one is the current.